### ISSUE TITLE
examples/eight_queens.tiger segfaults

### ISSUE THREAD

**Issue by tdp2110** [tdp2110/HaskellTiger#52]
but it compiles now!

**Comment by tdp2110** [comment]
no longer segfaults, but doesn't print anything :)

**Comment by tdp2110** [comment]
here's a simpler example that seems to have the same (current) problem:

```
let
    var N := 4
    type intArray = array of int
    var arr := intArray [ N ] of 0

    function printarray() =
        (print("[");
        for i := 0 to N-1
        do (print(itoa(arr[i]));
            print(" "));
        print("]\n"))

    function foo(c : int) =
        if c=N then printarray()
        else (arr[c] = c;
              foo(c+1))

in
    foo(0)
end
```

It should print `[0 1 2 3 ]` but prints `[0 0 0 0 ]`

**Comment by tdp2110** [comment]
the previous example is working correctly, actually, given the bug #59

**Comment by tdp2110** [comment]
OMG FIXED IN be4b387!
